cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -81,8 +82,8 @@
   private final Map<String, LocalizeResource> resourcesToLocalize;
   private final Supplier<MapReduceTaskContextProvider> taskContextProviderSupplier;
 
-  private InputFormatProvider inputFormatProvider;
-
+  private Map<String, InputFormatProvider> inputFormatProviders;
+  private HashMap<String, Class<? extends Mapper>> mapperClasses;
```

Comment:
But the InputFormatProvider and Mapper is one-to-one right? Why not use a class then? Using a Table seems just getting it more complicate.